### PR TITLE
Added additional check in withCookieHeader.

### PR DESCRIPTION
**Pull request by Miljan9602**
Added check for Host header. If check does not match domain from uri, cookie will not be added.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "guzzlehttp/guzzle",
+    "name": "miljan9602/guzzle",
```
Comment (Nyholm) [inline composer.json]: Oh, please dont rename the package =)

**Comment by nemanjan00** [comment]
We have no idea how exacly we would test for something like this. 

Problem happens when we want to send request to different reverse proxy, on purpose. 

We change address to `http://reverseproxyaddress/` and add header `Host: originalolaction` and, cookies for `reverseproxyaddress` are used, instead of those for `originallocation`

**Comment by GrahamCampbell** [comment]
I don't understand the use case here. Why would a cookie for a different domain be added?

**Comment by nemanjan00** [comment]
In some test cases, we want to pick specific reverse rpoxy instance. 

If we specify that reverse proxy in host, we need to specify Domain header, so reverse proxy knows what domain we are requesting, and, we also want to use cookies for that domain.

**Comment by Nyholm** [comment]
Friendly ping

**Comment by GrahamCampbell** [comment]
This is also missing tests.,

File: src/Cookie/CookieJar.php [hunk 2]
```diff
@@ -279,10 +279,15 @@ public function withCookieHeader(RequestInterface $request)
         $scheme = $uri->getScheme();
         $host = $uri->getHost();
         $path = $uri->getPath() ?: '/';
+        $headerHost = null;
+
+        if (sizeof($request->getHeader('Host')) > 0) {
```
Comment (GrahamCampbell) [inline src/Cookie/CookieJar.php:284]: When can there be no Host header?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Pull request closed without merging**